tests/network/primary_pod_network.go

```diff
@@ -0,0 +1,137 @@
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+)
+
+var _ = SIGDescribe("Primary Pod Network", func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+	})
+
+	Describe("Status", func() {
+		AssertReportedIP := func(vmi *v1.VirtualMachineInstance) {
+			By("Getting pod of the VMI")
+			vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+
+			By("Making sure IP reported on the VMI matches the one on the pod")
+			Expect(vmi.Status.Interfaces[0].IP).To(Equal(vmiPod.Status.PodIP), "The address reported in VMI's IP should match the PodIP")
+			Expect(vmi.Status.Interfaces[0].IPs).NotTo(BeEmpty(), "VMI should report a list of its IP addresses")
+			Expect(vmi.Status.Interfaces[0].IPs[0]).To(Equal(vmiPod.Status.PodIP), "The first address reported in VMI's IPs should match the PodIP")
+		}
+
+		Context("VMI connected to the pod network using the default (implicit) binding", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmi = setupVMI(virtClient, vmiWithDefaultBinding())
+			})
+
+			AfterEach(func() {
+				cleanupVMI(virtClient, vmi)
+			})
+
+			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
+		})
+
+		Context("VMI connected to the pod network using bridge binding", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmi = setupVMI(virtClient, vmiWithBridgeBinding())
+			})
+
+			AfterEach(func() {
+				cleanupVMI(virtClient, vmi)
+			})
+
+			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
+		})
+
+		Context("VMI connected to the pod network using masquerade binding", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmi = setupVMI(virtClient, vmiWithMasqueradeBinding())
+			})
+
+			AfterEach(func() {
+				cleanupVMI(virtClient, vmi)
+			})
+
+			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
+		})
+	})
+})
+
+func setupVMI(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
+	By("Creating the VMI")
+	var err error
+	vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+	Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+
+	By("Waiting until the VMI gets ready")
+	vmi = tests.WaitUntilVMIReady(vmi, tests.LoggedInAlpineExpecter)
+
+	return vmi
+}
+
+func cleanupVMI(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) {
+	if vmi != nil {
+		By("Deleting the VMI")
+		Expect(virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.GetName(), &metav1.DeleteOptions{})).To(Succeed())
```

Comment:
I would suggest waiting for the deletion to succeed.
But it may not be a real problem because we randomize the VMI names, the only potential problem may be with resources side effects (leaving them behind and maybe trying to use them in the next test).

Reply:
Right. Actually it is a real issue we faced in other tests where we overloaded the system with terminating VMs.